exchanges/orderbook/orderbook_test.go

```diff
@@ -26,6 +27,43 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+func TestGetAveragePrice(t *testing.T) {
+	var b Base
+	b.Exchange = "Binance"
+	cp, err := currency.NewPairFromString("ETH-USDT")
+	if err != nil {
+		t.Error(err)
+	}
+	b.Pair = cp
+	b.Bids = []Item{}
+	_, err = b.GetAveragePrice(false, 5)
+	if errors.Is(errNotEnoughLiquidity, err) {
+		t.Error("expected: %w, received %w", errNotEnoughLiquidity, err)
+	}
+	b = Base{}
+	b.Pair = cp
+	b.Asks = []Item{
+		{Amount: 5, Price: 4},
+		{Amount: 5, Price: 3},
+		{Amount: 5, Price: 2},
+		{Amount: 5, Price: 1},
+	}
+	avgPrice, err := b.GetAveragePrice(true, 15)
```

Comment:
ASK should also be ascending not descending. Can we also add in a test case for selling. Also, I am a little confused when I want to buy 15 ETH I don't expect the price to be 3 (in this context).  I expect the cost to be an average price to be $15 USDT. (forgive me if I am completely wrong.)